### PR TITLE
[KYUUBI #5565][AUTHZ] Support Delete/Insert/Update table command for Delta Lake

### LINKED ISSUE #5565: [Subtask] Support Delete/Insert/Update table command for Delta Lake
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues) and found no similar issues.

### Describe the subtask

Support Delete/Insert/Update table command for Delta Lake in Authz plugin.

### Parent issue

#5477 

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by zml1206**
### _Why are the changes needed?_
To close #5565.
- Support Delete from table command for Delta Lake in Authz.
- Support Insert table command for Delta Lake in Authz.
- Support Update table command for Delta Lake in Authz.
- Reduce the fields of `createTableSql`.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
No.

### CODE REVIEW

**Comment by zml1206** [comment]
cc @yaooqinn @bowenliang123

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -189,6 +163,62 @@ class DeltaCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
...
+    }
+  }
+
+  test("update table") {
+    withCleanTmpResources(Seq((s"$namespace1.$table1", "table"), (s"$namespace1", "database"))) {
+      doAs(admin, sql(s"CREATE DATABASE IF NOT EXISTS $namespace1"))
+      doAs(admin, sql(createTableSql(namespace1, table1)))
+      interceptContains[AccessControlException](
+        doAs(
+          someone,
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala:216]: shall we also do as with admin?

Comment (zml1206) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/DeltaCatalogRangerSparkExtensionSuite.scala]: Done

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thanks @zml1206,  merged to master

**Comment by zml1206** [comment]
Thanks, @yaooqinn

**Pull request closed without merging**